From 9b2d8612f0e9f029a1b18ce4210e302be8d91011 Mon Sep 17 00:00:00 2001
From: Akshat Lakhera <akshatlnct2024@gmail.com>
Date: Mon, 28 Sep 2026 04:20:00 +0200
Subject: [PATCH] welcome: inform user that installation process is completely
 offline

Inform users on the initial Welcome screen that the Qubes OS installer is
completely offline, does not require Internet access, and does not load
networking drivers.

Fixes QubesOS/qubes-issues#10213
---
 pyanaconda/ui/gui/spokes/welcome.glade | 25 ++++++++++++++++++++++---
 pyanaconda/ui/gui/spokes/welcome.py    |  8 ++++++++
 2 files changed, 30 insertions(+), 3 deletions(-)

--- a/pyanaconda/ui/gui/spokes/welcome.glade
+++ b/pyanaconda/ui/gui/spokes/welcome.glade
@@ -90,6 +90,26 @@
                   </packing>
                 </child>
                 <child>
+                  <object class="GtkLabel" id="offlineNoticeLabel">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="margin_left">36</property>
+                    <property name="margin_right">36</property>
+                    <property name="margin_bottom">6</property>
+                    <property name="label" translatable="yes">The Qubes OS installer is completely offline. It does not require Internet access and does not load any networking drivers.</property>
+                    <property name="wrap">True</property>
+                    <property name="xalign">0</property>
+                    <attributes>
+                      <attribute name="font-desc" value="Cantarell Italic 11"/>
+                    </attributes>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="position">1</property>
+                  </packing>
+                </child>
+                <child>
                   <object class="GtkLabel" id="pickLanguageLabel">
                     <property name="height_request">45</property>
                     <property name="visible">True</property>
@@ -107,7 +127,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">False</property>
-                    <property name="position">1</property>
+                    <property name="position">2</property>
                   </packing>
                 </child>
                 <child>
@@ -273,7 +293,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">True</property>
-                    <property name="position">2</property>
+                    <property name="position">3</property>
                   </packing>
                 </child>
               </object>
--- a/pyanaconda/ui/gui/spokes/welcome.py
+++ b/pyanaconda/ui/gui/spokes/welcome.py
@@ -207,6 +207,13 @@
 
         welcomeLabel.set_text(_("WELCOME TO %(name)s %(version)s.") %
                 {"name" : get_product_name().upper(), "version" : get_product_version()})
+
+        offlineNoticeLabel = self.builder.get_object("offlineNoticeLabel")
+        if offlineNoticeLabel:
+            offlineNoticeLabel.set_text(_(
+                "The Qubes OS installer is completely offline. "
+                "It does not require Internet access and does not load any networking drivers."
+            ))
 
         # Retranslate the language (filtering) entry's placeholder text
         languageEntry = self.builder.get_object("languageEntry")
-- 
2.45.2
